torchgeo/datasets/l8biome.py

```diff
@@ -0,0 +1,190 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""L8Biome dataset."""
+
+import abc
+import glob
+import hashlib
+import os
+from functools import lru_cache
+from typing import Any, Callable, Dict, List, Optional, cast
+
+import matplotlib.pyplot as plt
+import numpy as np
+import torch
+from matplotlib.colors import ListedColormap
+from PIL import Image
+from rasterio.crs import CRS
+from torch import Tensor
+from torch.utils.data import Dataset
+from .utils import check_integrity
+
+from .geo import NonGeoDataset, RasterDataset
+from .utils import BoundingBox, download_url, extract_archive, working_dir
+
+
+class L8Biome(RasterDataset):
+    r"""L8 Biome datasets.
+
+    The L8 Biome dataset `<https://landsat.usgs.gov/landsat-8-cloud-cover-assessment-validation-data>`__ (Landsat 8 Cloud Cover Assessment Validation Data) 
+    is a cloud validation dataset of Pre-Collection Landsat 8 Operational Land Imager (OLI) Thermal Infrared Sensor (TIRS) terrain-corrected (Level-1T) scenes.
+    
+    Dataset features:
+
+    * images evenly divided between eight unique biomes
+    * three cloud cover categories
+
+    Dataset format:
+
+    * Each cloud mask is in ENVI binary format. Includes all bands from the original Landsat Level-1 data product (GeoTIFF), and its associated Level-1 metadata (MTL.txt file)
+
+    If you use this dataset in your research, please cite the following:
+
+    * doi:10.5066/F7251GDH
+    * 10.1016/j.rse.2017.03.026
+
+    .. versionadded:: 0.5
+    """
+
+    url = "https://huggingface.co/datasets/torchgeo/l8biome" # redistributed from https://landsat.usgs.gov/landsat-8-cloud-cover-assessment-validation-data
+    filenames_to_md5 = {
+        "barren": "0ff8e32511e3e62fce9a08bed88bebab",
+        "forest": "8adf6b7b33b65e16c0f692c227cd38de",
+        "grass_crops": "fd38f2b6ff00cce8cd14ea6db82f10f4",
+        "shrubland": "2de5cdeea89390f5908663dcab5ca869",
+        "snow_ice": "e431f0b8e8431f8b04a440bf5083a91c",
+        "urban": "b6d39858117940930bf6fa4d63488374",
+        "water": "8ce42b9f7f58f4d9f8fd6801aa894421",
+        "wetlands": "ccd193eb7509e262cbe9588ecb8eddb4"
+    }
+
+    cmap = {
+        0: (0,0,0),
+        64: (64,64,64),
+        128: (128,128,128),
+        192: (192,192,192),
+        255: (255,255,255),
+    }
+    
```

Comment:
You're missing filename_glob and filename_regex. Without those, this dataset won't actually locate any files. See https://torchgeo.readthedocs.io/en/stable/tutorials/custom_raster_dataset.html.